protobuf/Dockerfile

```diff
@@ -1,46 +1,59 @@
 ARG ALPINE_VERSION=3.12
-ARG GO_VERSION=1.14.6
-ARG GRPC_GATEWAY_VERSION=1.14.6
-ARG GRPC_JAVA_VERSION=1.30.2
+ARG GO_VERSION=1.15.6
 # gRPC core version that applies to C++, C#, Objective-C, PhP, Python, Ruby
-ARG GRPC_VERSION=1.30.2
-ARG PROTOC_GEN_GO_VERSION=1.4.2
-# v1.3.1, using the version directly does not work: "tar: invalid magic"
-ARG PROTOC_GEN_GOGO_VERSION=5628607bb4c51c3157aacc3a50f0ab707582b805
+ARG GRPC_VERSION=1.34.0
+ARG PROTOBUF_C_VERSION=1.3.3
+
+ARG PROTOC_GEN_GO_VERSION=1.4.3
+ARG PROTOC_GEN_GO_GRPC_VERSION=1.34.0
+ARG GRPC_JAVA_VERSION=1.34.1
+# v1.3.2, using the version directly does not work: "tar: invalid magic"
+ARG PROTOC_GEN_GOGO_VERSION=b03c65ea87cdc3521ede29f62fe3ce239267c1bc
 ARG PROTOC_GEN_LINT_VERSION=0.2.1
+ARG GRPC_GATEWAY_VERSION=2.1.0
 ARG UPX_VERSION=3.96
 
 
-FROM alpine:${ALPINE_VERSION} as protoc_base
-RUN apk add --no-cache build-base curl cmake autoconf libtool git zlib-dev linux-headers && \
-    mkdir -p /out
+FROM alpine:${ALPINE_VERSION} as protoc_builder
+RUN apk add --no-cache build-base curl automake autoconf libtool git zlib-dev linux-headers cmake ninja
 
+RUN mkdir -p /out
 
-FROM protoc_base as protoc_builder
 ARG GRPC_VERSION
-RUN apk add --no-cache automake && \
-    git clone --recursive --depth=1 -b v${GRPC_VERSION} https://github.com/grpc/grpc.git /grpc && \
+RUN git clone --recursive --depth=1 -b v${GRPC_VERSION} https://github.com/grpc/grpc.git /grpc && \
     ln -s /grpc/third_party/protobuf /protobuf && \
-    cd /protobuf && \
+    mkdir -p /grpc/cmake/build && \
+    cd /grpc/cmake/build && \
+    cmake \
+        -GNinja \
+        -DBUILD_SHARED_LIBS=ON \
+        -DCMAKE_INSTALL_PREFIX=/usr \
+        -DCMAKE_BUILD_TYPE=Release \
+        -DgRPC_INSTALL=ON \
+        -DgRPC_BUILD_TESTS=OFF \
+        ../.. && \
+    cmake --build . --target plugins && \
+    cmake --build . --target install && \
+    DESTDIR=/out cmake --build . --target install 
```

Comment:
Just curious: are these changes dictated by the new version of grpc? Did they move from make to cmake?

Reply:
Yes they now only use cmake and bazel for building.